GoogleForCreators/web-stories-wp
# PPC: Single Issue Card Component

## Context

First installment of new card component for use in PPC v2.0 

## Summary

Adds a new card component to editor. 
<img width="933" alt="Screen Shot 2021-06-17 at 10 25 01 AM" src="https://user-images.githubusercontent.com/10720454/122447218-5da38000-cf58-11eb-8989-b20224709e58.png">


## Relevant Technical Choices

- Decided to just use a css grid with named columns so that card structure is flexible. The template grid area related to a `CARD_TYPE` is returned to the `Container` and voila, a card! 
- Added a list component to the design system typography.
- ChecklistCard has a storybook demo for every [single issue card variant in figma](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=8378%3A3528)
- Made the `Title` area conditionally render as a button for the functionality currently in the checklist where you can sometimes (most of the time) click the title area and have the spot to fix highlight, this prevents us from needing to set a tab index or a keyDown event. 

If ya'll don't like having `helper`, `cta`, and `thumbnail` as props that just pass the contents of those sections through the card we can ditch the overall `ChecklistCard` and pull the individual styled components in separate by checklist issue. To me, this makes it clear what all composes a card. Plus, I thought it would make variants easier for when there are more than 1 thumbnail to an issue. But really it's 6 or half dozen. 

## To-do

- Need to talk to design about the card hover bg conflicting with the CTA hover. 

## User-facing changes

None, just some storybook stuff.

## Testing Instructions

Spin up storybook locally and see the checklistCard in the stories-editor section. 


### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #7914 
Partially addresses #7915
